src/components/core/GeoChart.tsx

```diff
@@ -70,39 +103,136 @@ export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
         );
     }
 
+    public onLegendItemClick(item: any) {
+        this.setState({
+            legendIndexEnabled: set<boolean[]>(
+                [...this.state.legendIndexEnabled],
+                item.legendIndex,
+                !this.state.legendIndexEnabled[item.legendIndex],
+            ),
+            legendNamesEnabled: this.getSelectedSegmentItem(
+                item.name,
+                !this.state.legendIndexEnabled[item.legendIndex],
+            ),
+        });
```

Comment:
be careful when updating state using current state's value
https://www.rockyourcode.com/react-set-state-with-prev-state-and-object-spread-operator/

Reply:
fixed